pyigm/field/igmfield.py

```diff
@@ -0,0 +1,236 @@
+""" Class for correlating galaxies and IGM within a given field on the sky
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+import warnings
+import pdb
+
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+from astropy import constants as const
+
+#from xastropy.xutils import xdebug as xdb
+
+
+class IgmGalaxyField(object):
+    """ Class for a field associating galaxies to the IGM/CGM
+
+    Parameters
+    ----------
+    radec : tuple or SkyCoord
+        (RA,DEC) in deg or astropy.coordinate
+    name : str, optional
+    """
+
+    # Initialize 
+    def __init__(self, radec, name='', cosmo=None, verbose=False):
+        # coord
+        if isinstance(radec, (tuple)):
+            self.coord = SkyCoord(ra=radec[0], dec=radec[1])
+        elif isinstance(radec, SkyCoord):
+            self.coord = radec
+
+        # Field
+        self.name = name
```

Comment:
How about coming up with a naming convention for a field, e.g. using the central coordinates? (similar to the current name in the Galaxy class). Use this naming convention as default in case name is not given explicitly.